core/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java

```diff
@@ -230,6 +238,7 @@ public void readFrom(StreamInput in) throws IOException {
         type = in.readString();
         id = in.readString();
         routing = in.readOptionalString();
+        parent = in.readOptionalString();
```

Comment:
you would need to protect this read with a version check so that it does not fail when reading data coming from an old node - this would look like this:

```java
if (in.getVersion().onOrAfter(Version.V_2_2_0)) {
  parent = in.readOptionalString();
} else {
  parent = null;
}
```